pkg/controllers/data.go

```diff
@@ -0,0 +1,147 @@
+package controllers
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+
+	. "github.com/CESARBR/knot-cloud-storage/pkg/entities"
+	"github.com/CESARBR/knot-cloud-storage/pkg/interactor"
+	"github.com/gorilla/mux"
+)
+
+type DataController struct {
+	DataInteractor interactor.DataInteractor
+}
+
+type DataInterface interface {
+	GetAll()
+	GetByID()
+	Create()
+}
+
+type errorMessage struct {
+	error   bool
+	message string
+}
+
+func NewDataController(dataInteractor interactor.DataInteractor) DataController {
+	return DataController{dataInteractor}
+}
+
+func respondWithError(w http.ResponseWriter, code int, msg string) {
+	respondWithJson(w, code, map[string]string{"message": msg})
+}
+
+func respondWithJson(w http.ResponseWriter, code int, payload interface{}) {
+	response, _ := json.Marshal(payload)
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(code)
+	w.Write(response)
+}
+
+func (d *DataController) GetAll(w http.ResponseWriter, r *http.Request) {
+
```

Comment:
Keep an eye out for excessive blank lines, only use them when [necessary](https://about.sourcegraph.com/handbook/engineering/go_style_guide#use-new-lines-intentionally) :stuck_out_tongue:

Reply:
Done!